Dockerfile

```diff
@@ -13,8 +13,8 @@ WORKDIR /usr/src/app
 COPY .ruby-version .
 
 # Install Rbenv & Ruby
-RUN git clone https://github.com/rbenv/rbenv.git ${RBENV_ROOT} && \
-    git clone https://github.com/rbenv/ruby-build.git ${RBENV_ROOT}/plugins/ruby-build && \
+RUN git clone --depth 1 --branch v1.1.2 https://github.com/rbenv/rbenv.git ${RBENV_ROOT} && \
+    git clone --depth 1 --branch v20200520 https://github.com/rbenv/ruby-build.git ${RBENV_ROOT}/plugins/ruby-build && \
```

Comment:
there's the downside of having to update these versions later, I wonder if this makes the process significantly faster. If it does, let's keep this :+1:

Reply:
Cloning a specific release with `--branch` does have the downside of having to change it when we need to update to the latest release. On the other side, this makes the installation of those tools reproducible since we're always building the same version.

As for the speed, it's not relevant when we're talking about `--branch`. This is when `--depth 1` comes into play. I've put below some numbers after timing the git clone commands with `time`.

For ruby-build:
```
$ time git clone --quiet https://github.com/rbenv/ruby-build.git 
0.50s user 0.14s system 15% cpu 4.053 total (~4 seconds)

$ time git clone --quiet --depth 1 https://github.com/rbenv/ruby-build.git
0.06s user 0.05s system 6% cpu 1.785 total (~1.7 seconds)
```

For rbenv:
```
$ time git clone --quiet https://github.com/rbenv/rbenv.git 
0.15s user 0.07s system 15% cpu 1.421 total (~1.4 seconds)

$ time git clone --quiet --depth 1 https://github.com/rbenv/rbenv.git 
0.05s user 0.05s system 6% cpu 1.438 total (~1.4 seconds)
```

So as we can see here, using `--depth 1` does make the cloning of `ruby-build` a bit faster. For `rbenv`, it's pretty much the same since the repository doesn't have many commits.

Reply:
ah, ok, I agree, it's better to always use the same version :+1: